script/import_scripts/base.rb

```diff
@@ -430,7 +430,7 @@ def create_categories(results)
         # make sure categories don't go more than 2 levels deep
         if params[:parent_category_id]
           top = Category.find_by_id(params[:parent_category_id])
-          top = top.parent_category while top && !top.parent_category.nil?
+          top = top.parent_category while (top&.height_of_ancestors || -1) + 1 >= SiteSetting.max_category_nesting
```

Comment:
I think the condition is inverted - this should be `proposed_nesting <= max_category_nesting`

Reply:
No, that is correct. I just tried this again and it is all working as expected.

The idea is to ensure that the parent category (top) is on a maximum level of max_category_nesting - 1.